pkg/ruby/discover_test.go

```diff
@@ -0,0 +1,144 @@
+package ruby
+
+import (
+	"testing"
+
+	"chainguard.dev/apko/pkg/build/types"
+	"chainguard.dev/melange/pkg/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/wolfi-dev/wolfictl/pkg/melange"
+)
+
+func TestIsRubyPackage(t *testing.T) {
+	testConfig := config.Configuration{
+		Environment: types.ImageConfiguration{
+			Contents: types.ImageContents{
+				Packages: []string{
+					"",
+				},
+			},
+		},
+	}
+	o := Options{
+		RubyVersion: "3.2",
+	}
+
+	shouldBeTrue := []string{
+		"ruby-3.2",
+		"ruby-3.2-dev",
+	}
+	for _, pkg := range shouldBeTrue {
+		testConfig.Environment.Contents.Packages[0] = pkg
+		assert.Truef(t, o.isRubyPackage(testConfig), "%s should be true", pkg)
+	}
+
+	shouldBeFalse := []string{
+		"ruby-2.2",
+		"ruby-3.22",
+		"ruby-33.2",
+		"ruby-2.2-dev",
+		"ruby-3.22-dev",
+		"ruby-33.2-dev",
+	}
+	for _, pkg := range shouldBeFalse {
+		testConfig.Environment.Contents.Packages[0] = pkg
+		assert.Falsef(t, o.isRubyPackage(testConfig), "%s should be false", pkg)
+	}
+}
```

Comment:
This might make a good case to use table-based tests

Reply:
Good note! Converted